sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -193,7 +193,16 @@ class SessionCatalog(
   def getDatabaseMetadata(db: String): CatalogDatabase = {
     val dbName = formatDatabaseName(db)
     requireDbExists(dbName)
-    externalCatalog.getDatabase(dbName)
+    val database = externalCatalog.getDatabase(dbName)
+
+    // default database's location always use the warehouse path,
+    // and since the location of database stored in metastore is qualified,
+    // here we also make qualify for warehouse location
+    val dbLocation = if (dbName == SessionCatalog.DEFAULT_DATABASE) {
+      SessionCatalog.makeQualifiedPath(conf.warehousePath, hadoopConf).toString
+    } else database.locationUri
```

Comment:
Actually, maybe we can write a separate function for getting the location of default database. Just put it below the function [`getDefaultDBPath`](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala#L227-L234). We might need to rename `getDefaultDBPath`. The name looks confusing now.